backend/clubs/serializers.py

```diff
@@ -1409,10 +1409,23 @@ class UserClubVisitSerializer(serializers.ModelSerializer):
 
     person = serializers.HiddenField(default=serializers.CurrentUserDefault())
     club = ClubListSerializer(read_only=True)
+    ip = serializers.HiddenField(default=None)
+
+    def save(self):
+        self.validated_data["ip"] = self.get_ip()
+        return super().save()
+
+    def get_ip(self):
+        x_forwarded_for = self.context["request"].META.get("HTTP_X_FORWARDED_FOR", None)
+        if x_forwarded_for:
+            ip = x_forwarded_for.split(",")[0]
+        else:
+            ip = self.context["request"].META.get("REMOTE_ADDR", None)
+        return ip
 
     class Meta:
         model = ClubVisit
-        fields = ("club", "visit_type", "person")
+        fields = ("club", "visit_type", "person", "ip")
```

Comment:
I'm pretty sure you can leave out the `"ip"` and the HiddenField altogether, and it will do the same thing. You'll need to test this, though.

Reply:
I double checked this and yes it did the same thing! Thanks for pointing that out